lib/rubocop/cop/rspec/multiple_memoized_helpers.rb

```diff
@@ -0,0 +1,139 @@
+      #
+      # This cop is configurable using the `Max` option and the `AllowSubject`
+      # which will configure the cop to only register offenses on calls to
+      # `let` and not calls to `subject`.
+      #
+      # @example
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #     let!(:booger) { [] }
+      #     subject { {} }
+      #     subject(:wat) { {} }
+      #     subject!(:boo) { {} }
+      #   end
+      #
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #
+      #     context 'when stuff' do
+      #       let!(:booger) { [] }
+      #       subject { {} }
+      #       subject(:wat) { {} }
+      #       subject!(:boo) { {} }
+      #     end
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let!(:booger) { [] }
+      #     subject { {} }
+      #     subject(:wat) { {} }
+      #     subject!(:boo) { {} }
+      #   end
+      #
+      #   describe MyClass do
+      #     context 'when stuff' do
+      #       let(:foo) { [] }
+      #       let(:bar) { [] }
+      #       let!(:booger) { [] }
+      #     end
+      #
+      #     context 'when other stuff' do
+      #       subject { {} }
+      #       subject(:wat) { {} }
+      #       subject!(:boo) { {} }
+      #     end
+      #   end
+      #
+      # @example with AllowSubject configuration
+      #
+      #   # rubocop.yml
+      #   # RSpec/MultipleMemoizedHelpers:
+      #   #   AllowSubject: true
+      #
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #     let!(:booger) { [] }
+      #     let(:subject) { {} }
+      #     let(:wat) { {} }
+      #     let!(:boo) { {} }
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #     let!(:booger) { [] }
+      #     subject { {} }
+      #     subject(:wat) { {} }
+      #     subject!(:boo) { {} }
+      #   end
+      #
+      # @example with Max configuration
+      #
+      #   # rubocop.yml
+      #   # RSpec/MultipleMemoizedHelpers:
+      #   #   Max: 0
+      #
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     def foo; []; end
+      #   end
+      #
+      class MultipleMemoizedHelpers < Base
+        MSG = 'Example group has too many memoized helpers [%<count>d/%<max>d]'
+
+        def on_block(node)
+          return unless spec_group?(node)
+
+          count = total_helpers(node)
+
+          return if count <= max
+
+          add_offense(node, message: format(MSG, count: count, max: max))
+        end
+
+        private
+
+        def total_helpers(node)
+          helpers(node) +
+            node.each_ancestor(:block).map(&method(:helpers)).sum
```

Comment:
This way we count overrides in nested context which is quite a valid usage for `let`. I'd rather count the helpers on a single level instead. Or you'll have to map them to names and count uniq names

Reply:
Done!